cdk/lib/index.ts

```diff
@@ -0,0 +1,307 @@
+      stageValues: {
+        PROD: "prod",
+        CODE: "code",
+      }
+    });
+
+    const typerighterBucketName = 'typerighter';
+
+    const dbPort = 5432;
+
+    // Rule manager app
+
+    const ruleManagerAppName = "typerighter-rule-manager";
+
+    const ruleManagerDomain = this.withStageDependentValue({
+      app: ruleManagerAppName,
+      variableName: "ruleManagerDomain",
+      stageValues: {
+        PROD: managerDomainPROD,
+        CODE: managerDomainCODE,
+      }
+    });
+
+    const ruleManagerApp = new GuPlayApp(this, {
+      app: ruleManagerAppName,
+      instanceType: new InstanceType("t4g.micro"),
+      userData: `#!/bin/bash -ev
+        aws --quiet --region ${this.region} s3 cp s3://composer-dist/${this.stack}/${this.stage}/typerighter-rule-manager/typerighter-rule-manager.deb /tmp/package.deb
+        dpkg -i /tmp/package.deb`,
+      access: {
+        scope: AccessScope.PUBLIC,
+      },
+      certificateProps: {
+        CODE: { domainName: managerDomainCODE },
+        PROD: { domainName: managerDomainPROD },
+      },
+      monitoringConfiguration: {
+        noMonitoring: true,
+      },
+      roleConfiguration: {
+        additionalPolicies: [pandaAuthPolicy],
+      },
+    });
+
+    const ruleManagerDnsRecord = new GuDnsRecordSet(this, "manager-dns-records", {
+      name: ruleManagerDomain,
+      recordType: RecordType.CNAME,
+      resourceRecords: [ruleManagerApp.loadBalancer.loadBalancerDnsName],
+      ttl: Duration.minutes(60)
+    });
+
+    // Checker app
+
+    const checkerAppName = "typerighter-checker";
+
+    const checkerDomain = this.withStageDependentValue({
+      app: checkerAppName,
+      variableName: "checkerDomain",
+      stageValues: {
+        PROD: checkerDomainPROD ,
+        CODE: checkerDomainCODE,
+      },
+    });
+
+    const checkerApp = new GuPlayApp(this, {
+      app: checkerAppName,
+      instanceType: new InstanceType("t4g.small"),
+      userData: `#!/bin/bash -ev
+mkdir /etc/gu
+
+cat > /etc/gu/tags << 'EOF'
+Stage=${this.stage}
+Stack=${this.stack}
+App=typerighter-checker
+EOF
+
+cat > /etc/gu/typerighter-checker.conf << 'EOF'
+include "application"
+typerighter.ngramPath="/opt/ngram-data"
+EOF
+
+aws --quiet --region ${this.region} s3 cp s3://composer-dist/${this.stack}/${this.stage}/typerighter-checker/typerighter-checker.deb /tmp/package.deb
+dpkg -i /tmp/package.deb`,
+      access: {
+        scope: AccessScope.PUBLIC,
+      },
+      certificateProps: {
+        CODE: { domainName: checkerDomainCODE },
+        PROD: { domainName: checkerDomainPROD },
+      },
+      monitoringConfiguration: {
+        noMonitoring: true,
+      },
+      roleConfiguration: {
+        additionalPolicies: [
+          pandaAuthPolicy,
+          new GuPutCloudwatchMetricsPolicy(this),
+        ],
+      },
+    });
+
+
+    const typerighterBucket = new GuS3Bucket(this, "typerighter-bucket", {
+      bucketName: typerighterBucketName,
+    });
```

Comment:
This won't work quite the way we'd expect! As we're explicitly setting the bucket name, we're creating a scenario where CODE and PROD will _both_ attempt to create a bucket of this name. This isn't legal as the bucket name is a unique identifier, thus can only be created once.

Alternatives would be to create a bucket per stack (like we were previously doing in this branch) or create the bucket outside of this stack.

Apologies, as this is the opposite of what was previously recommended. A Discussion has been opened [here](https://github.com/guardian/recommendations/discussions/74).

Reply:
Ah, yep! Should've spotted – what's best? Can't help but feel that two buckets are preferable, although not ideal, as they keep everything as IaC, but keen to hear your take.

Reply:
Amended to `typerighter-${stage}` per the way that discussion is going – I do prefer keeping bucket creation in VCS.